crates/sui-core/src/narwhal_manager/mod.rs

```diff
@@ -140,13 +153,13 @@ impl NarwhalManager {
             now.elapsed().as_secs_f64()
         );
 
-        *running = Running::True(committee.epoch());
+        *worker_running = Running::True(committee.epoch());
     }
 
     // Shuts down whole Narwhal (primary & worker(s)) and waits until nodes
     // have shutdown.
     pub async fn shutdown(&self) {
-        let mut running = self.running.lock().await;
+        let mut running = self.primary_running.lock().await;
```

Comment:
I see that we handle the primary mutex here, but what about the `worker` one? We never seem to set it to `false` when we do shutdown

Reply:
this was a mistake, fixed!